### ISSUE TITLE
Android studio flutter plugin should use --config-only when opening example app in plugin

### ISSUE THREAD

**Issue by reidbaker** [flutter/flutter-intellij#6898]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

https://github.com/flutter/flutter/issues/19830 Is an issue with 85 upvotes where users are confused about how they can get their code to link correctly in android studio. We also have a documentation update task but this request is to avoid a common error. 

Plugins are not buildable on their own so users already need to take the unintuitive step to open the example app we suggest in each plugin. The problem with that is we generate files on demand during a `flutter run` or `flutter build` that are required for android studio to know how to link the source. 

--config-only is a build step we added to generate the gradle files required. 

### Proposal

Modify the flutter android studio plugin to mitigate this class of errors for users by either building or running config-only for them. 

This is important despite the migration to vs code as our suggested editor because we still tell plugin developers working on android or add to app developers to use android studio.

**Comment by AlexV525** [comment]
Do you mean `flutter build apk --config-only` before opening the example? If I get it right, this would add a significant waiting duration, and we also need to determine whether the example needs to be built with config.

**Comment by reidbaker** [comment]
I don't think it would be blocking open but we would run that command when a plugin was opened and did not already have gradlew files. That is the determination for if we need to config. 

Full builds also do the work of configuration the config flag just stops after configuration. Android studio relies on gradle files to know what projects to link similar to our pubspec and when they are missing android studio does gives the user lots of errors that make work difficult.

**Comment by reidbaker** [comment]
Would a full build make more sense instead of just configuring the plugin and example app?

**Comment by AlexV525** [comment]
I'm not sure about if we should build the artifact or the config. I've checked the description in https://github.com/flutter/flutter/issues/19830 and it seems the original question is they opened the `plugin/android` instead of `plugin/example/android` which results in errors. So I tried to open an example without `gradlew` and it seems working well after the sync process is done.

![image](https://github.com/flutter/flutter-intellij/assets/15884415/bd8a6ed1-180a-421c-89c4-3f68d69c8e20)
